classes/Dispatcher.php

```diff
@@ -373,20 +373,19 @@ public function dispatch()
 
             // Dispatch module controller for front office
             case self::FC_MODULE:
-                $module_name = Validate::isModuleName(Tools::getValue('module')) ? Tools::getValue('module') : '';
-                $module = Module::getInstanceByName($module_name);
+                $module = Validate::isModuleName(Tools::getValue('module')) ? Module::getInstanceByName(Tools::getValue('module')) : false;
```

Comment:
Hi @PrestaEdit 
I think `Validate::isModuleName` is redondant here because `Module::getInstanceByName` already check that.

Reply:
And it's especially why this change is here. As the second method will check it and throw an error, you have un uncaught error instead of a page not found, for example, in a dev environment.

Reply:
I have seen that this only happens when we are in development mode, so, so I think this is the expected behavior so the dev will be able to see the error instead of redirecting to a 404 page.